.github/workflows/publish.yml

```diff
@@ -0,0 +1,57 @@
+name: Build and publish SDK package
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+
+permissions:
+  contents: read
+  packages: write
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: self-hosted # ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: package-tag
+        name: Determine tag for package
+        shell: bash
+        run: |
+          if [ "${{ github.event_name }}" = "schedule" ]; then
+            NPM_TAG="nightly"
+          elif [[ $GITHUB_REF == refs/heads/develop ]]; then
+            NPM_TAG="next"
+          elif [[ $GITHUB_REF == refs/tags/v*-* ]]; then
+            NPM_TAG=next
+          elif [[ $GITHUB_REF == refs/tags/v* ]]; then
+            NPM_TAG=latest
+          fi
+          echo "NPM_TAG=${NPM_TAG}" >> $GITHUB_ENV
+          echo "npm_tag=${NPM_TAG}" >> $GITHUB_OUTPUT
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+      - name: Enable Corepack
+        run: corepack enable
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          registry-url: 'https://npm.pkg.github.com'
+          scope: '@momentum-xyz'
+      - id: npm-deps
+        name: Install dependencies
+        run: npm ci --ignore-scripts
+      - id: build
+        name: NPM build
+        run: npm run build
+      - id: publish
+        name: Publish NPM package
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npm publish --tag "${{ steps.npm-package-tag.outputs.npm_tag }}"
```

Comment:
which repo? there is no regisrty configured in the package.json. So this would try to upload to npm.com